EventGenerator/src/RPCGun_module.cc

```diff
@@ -56,36 +54,45 @@ namespace mu2e {
 
   //================================================================
   class RPCGun : public art::EDProducer {
-    fhicl::ParameterSet psphys_;
-
-    BinnedSpectrum spectrum_;
-
-    int                 verbosityLevel_;
-    int                 generateInternalConversion_;
-    bool applySurvivalProbability_;
-    double survivalProbScaling_;
-    double tmin_;
-
-    double              czmin_;
-    double              czmax_;
-    double              phimin_;
-    double              phimax_;
+  public:
+    struct Config {
+      using Name=fhicl::Name;
+      using Comment=fhicl::Comment;
+        fhicl::Atom<art::InputTag> inputSimParticles{Name("inputSimParticles"),Comment("A SimParticleCollection with input stopped muons.")};
+        fhicl::Atom<unsigned> verbosity{Name("verbosity"),0};
+        fhicl::Atom<std::string> RPCType{Name("RPCType"),Comment("a process code, should be either RPCInternal or RPCExternal") };
+        fhicl::DelegatedParameter spectrum{Name("spectrum"), Comment("Parameters for BinnedSpectrum")};
+        fhicl::Atom<bool> doHistograms{Name("doHistograms"),false}; 
+    };
+
+    using Parameters= art::EDProducer::Table<Config>;
+    explicit RPCGun(const Parameters& conf);
+
+    virtual void produce(art::Event& event) override;
+    void addParticles(StageParticleCollection* output,art::Ptr<SimParticle> pistop);
+    double MakeEventWeight(art::Ptr<SimParticle> p);
+    //----------------------------------------------------------------
+  private:
+    art::ProductToken<SimParticleCollection> const simsToken_;
+    unsigned verbosity_;
 
     art::RandomNumberGenerator::base_engine_t& eng_;
-
-    CLHEP::RandGeneral  randSpectrum_;
+    CLHEP::RandExponential randExp_;
     CLHEP::RandFlat     randomFlat_;
-    RandomUnitSphere    randomUnitSphere_;
-    PionCaptureSpectrum pionCaptureSpectrum_;
-
-    RootTreeSampler<IO::StoppedParticleTauNormF> stops_;
-    std::unique_ptr<ProtonPulseRandPDF>  protonPulse_;
-
+    std::string RPCType_;
+    BinnedSpectrum    spectrum_;
     bool doHistograms_;
-    fhicl::ParameterSet protonPset_;
-
-    double generateEnergy();
-
+    RandomUnitSphere   randomUnitSphere_;
+    CLHEP::RandGeneral randSpectrum_;
+
+    const double            tmin_ = -1;
```

Comment:
`tmin_` is still not used anywhere. Either its original purpose can be re-implemented (perhaps a check that `tstop > tmin_` was the purpose?) or it can probably be deleted

Reply:
deleted for now, might be needed when I extend this module later on